torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -391,6 +391,43 @@ TensorView* TensorView::rFactor(const std::vector<int>& axes) {
   return producer;
 }
 
+std::vector<TensorView*> TensorView::duplicate() {
+  FusionGuard fg(fusion());
+
+  TORCH_INTERNAL_ASSERT(
```

Comment:
Shouldn't this be a user error rather than an internal error as it's a contract of this function?